pandas/_libs/tslib.pyx

```diff
@@ -412,7 +454,7 @@ def array_with_unit_to_datetime(
                 oresult[i] = <object>NaT
             else:
                 try:
-                    oresult[i] = Timestamp(cast_from_unit(val, unit))
+                    oresult[i] = Timestamp(val, unit=unit)
                 except OverflowError:
                     oresult[i] = val
```

Comment:
would it work to just catch `ValueError` here too, to address
```
to_datetime([1.5], unit='M', errors="ignore")
```
raising?

Reply:
id like to punt on this for the time being.   i think this path will need a revamp to be more consistent with non-unit path anyway

Reply:
sure thing